### DISCUSSION TITLE
How To Jump From Assembler To A C Label

### DISCUSSION THREAD

**Discussion by wizardmanannan** [cc65/cc65#2472] [Q&A] [answered]
Hello,

I am having issues working out how to jump from an assembly file to a label defined in C.

Here is what I am doing.

There may be typos in my example, I just threw it together in the forum. It is the jumping back to C that I am interested in; the rest works in the real code.

On the C side I have a function like this:

`
extern void fooAssm();
void fooC()
{
  asm("jmp %v",  fooAssm);
  myLabelReturn:
}
`

On the assembly side I have:
`
.import _myLabelReturn
.export fooAssm
fooAssm:
jmp _myLabelReturn
`

Jumping to the assembly is easy, but jumping back to C results in a  'unresolved external myLabelReturn' error.

Why aren't I using jsr?
This is a cycle saving trick for a frequently used function. 

It saves 3 cycles to do a jump instead of a jsr. The function is always only called from the one place.

Thanks

**Comment by polluks** [comment, marked as answer]
Well, the linker is looking for functions not labels.
You did not export myLabelReturn because you could not do it properly, just think about its data type.
A C label is not visible in assembler at all, this is not how the interface works.

**Reply by wizardmanannan** [reply]
Thanks for your answer.

I will now try a different strategy.

Thanks
